### PR TITLE
Migrate to libcurl

### PR DESCRIPTION
**Pull request by wmcrobertsq**
1. Replace MeteringClientWrapper with CurlWrapper - which directly calls libcurl.
2. Remove all references to metering client.
3. Follow design patterns from QryptSecurity (performCurl) and original metering client code when implementing call to libcurl.
4. Add in .devcontainer and .vscode to make it easy for developers to get started.

Here is the successful pipeline run for this branch:
https://hard-carbon.visualstudio.com/Qrypt/_build/results?buildId=54290&view=results

### CODE REVIEW

File: src/lib/CurlWrapper.cpp [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+}
+
+CurlResponse CurlWrapper::performCURL(size_t goal) {
+
+    CurlResponse response = {};
+
+    CURL *curlHandle = curl_easy_init();
+
+    std::string fullURL = "https://api-eus.qrypt.com/api/v1/quantum-entropy?size=";
+    size_t goalInKib = goal / KB;
```
Comment (qryptdev) [inline src/lib/CurlWrapper.cpp:99]: Do we need to round up here?

**Review by qryptdev — COMMENTED** [review]
Some small comments. -Sam

Bill, I think you should remove devcontainer and vscode files from this PR and reupload them as two (small) separate PRs. I don't want the discussion of whether or not they belong in the repo to bog down your very nice changes :)

**Comment by wmcrobertsq** [comment]
> 

Removed.

File: src/lib/CMakeLists.txt [hunk 2]
```diff
@@ -1,18 +1,13 @@
...
+    CurlWrapper.cpp
     RandomBuffer.cpp
     log.cpp
     osmutex.cpp
     GlobalData.cpp
-    MeteringClientWrapper.cpp
     main.cpp)
 
-target_include_directories(qryptoki PUBLIC "../../inc" "metering-client")
+    target_include_directories(qryptoki PUBLIC "../../inc")
```
[the remark this answers is not in the document]
Comment (wmcrobertsq) [reply src/lib/CMakeLists.txt:11]: Took out indent.

File: src/lib/CMakeLists.txt [hunk 3]
```diff
@@ -1,18 +1,13 @@
...
 
-target_link_libraries(qryptoki PUBLIC dl curl)
+    target_link_libraries(qryptoki PUBLIC dl curl)
```
[the remark this answers is not in the document]
Comment (wmcrobertsq) [reply src/lib/CMakeLists.txt:13]: Took out indent.

File: src/lib/CurlWrapper.cpp [hunk 4]
```diff
@@ -0,0 +1,191 @@
+/**
+ * Edited from file of the same name in QDARACLI -Sam
```
[the remark this answers is not in the document]
Comment (wmcrobertsq) [reply src/lib/CurlWrapper.cpp:2]: Removed.

File: src/lib/CurlWrapper.cpp [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+            case 401: {
+                 DEBUG_MSG("Unauthorized, invalid Credentials or problem with the SSL CA cert.");
+                 return CKR_QRYPT_TOKEN_INVALID;
+            } break;
+            case 403: {
+                 DEBUG_MSG("Unknown server failure %zu", response.code);
+                 return CKR_QRYPT_TOKEN_OTHER_FAIL;
+            } break;
+            case 404: {
+                 DEBUG_MSG("Random pool is expired, cannot retrieve sample");
```
[the remark this answers is not in the document]
Comment (wmcrobertsq) [reply src/lib/CurlWrapper.cpp:71]: Removed.

**Comment by qryptdev** [comment]
Approve. -Sam

**Pull request merged**